SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/AbstractDeviceInformationHelper.cs

```diff
@@ -4,16 +4,20 @@
 using System;
 using System.Reflection;
 using System.Threading.Tasks;
-
 using Microsoft.AppCenter.Windows.Shared.Utils;
 
 namespace Microsoft.AppCenter.Utils
 {
     public abstract class AbstractDeviceInformationHelper : IDeviceInformationHelper
     {
-        public virtual async Task<Ingestion.Models.Device> GetDeviceInformationAsync()
+        public virtual Task<Ingestion.Models.Device> GetDeviceInformationAsync()
```

Comment:
Why making Async synchronous in reality? There is no value in having the async since it will block before even returning a task object.

Reply:
Also if we do go through with this change, the method name needs to change.

Reply:
No this returns a `Task` so nothing actually changes. It won't block before returning a `Task`. Have you looked at the UWP implementation? It is the reason we need to return a `Task`. But since this base implementation does not `await`, using `async` doesn't do anything.

Reply:
And it is still conventional to use the `Async` suffix when you return a `Task`, even if the keyword is not used. We also introduced a truly synchronous version that is needed for the crash module. That one doesn't block at all.

Reply:
We need to discuss it offline.

Reply:
So in UWP we override it to be real async, ok.

Since we have only 2 implementations it should really be only in windows desktop the task.fromresult, that would make the whole thing less confusing. So please make method abstract and implement only task.fromresult in windowsdesktop.

Reply:
Sounds good. But let's do that in another PR